src/generators.cpp

```diff
@@ -179,18 +171,21 @@ std::string to_string(DeviceType device_type) {
       return "WebGpu";
     case DeviceType::QNN:
       return "QnnWithSharedMemory";
+    default:
+      throw std::runtime_error("Unknown device type");
   }
-  throw std::runtime_error("Unknown device type");
```

Comment:
Is the static analysis tools we have smart enough to detect that the control will never reach end of this function. Do we need a dummy `return std::string()`?

Reply:
It was actually a build warning on android or iOS that spurred me to fix that. I didn't see any other issues after doing it.